### PR TITLE
Rename `ModelVersionOperation` enum in UC registry REST store

### PR DESCRIPTION
**Pull request by jerrylian-db**
## What changes are proposed in this pull request?

This PR fixes the following two small issues found the in UC model registry bug bash:
* The `ModelVersionOperation` enum has incompatible string names with the `uc-model-registry`.
* The API path for getting scoped tokens is incomplete.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

I tested the fix on a UC-enabled test shard. The client successfully retrieved the scoped token.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [x] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/artifact/unity_catalog_models_artifact_repo.py [hunk 1]
```diff
@@ -21,9 +28,7 @@
 
 from mlflow.store._unity_catalog.registry.utils import get_artifact_repo_from_storage_info
 
-REGISTRY_GET_SCOPED_TOKEN_ENDPOINT = (
-    "/mlflow/unity-catalog/model-versions/generate-temporary-credentials"
-)
+_METHOD_TO_INFO = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
```
Comment (jerrylian-db) [inline mlflow/store/artifact/unity_catalog_models_artifact_repo.py:31]: Using this instead because I find it more reliable than hardcoding the endpoint

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/store/artifact/unity_catalog_models_artifact_repo.py (hunk 1 above)
Comment (smurching) [reply mlflow/store/artifact/unity_catalog_models_artifact_repo.py:29]: Hm @jerrylian-db question on this PR, how come the enum encoding fails in our case and not for others in OSS MLflow? For example [check out the code](https://github.com/mlflow/mlflow/blob/2e9c982b6f9a010c744b2e3d1a3a1c1203538cb3/mlflow/store/tracking/rest_store.py#L137-L149) for UpdateRun where we pass `RunStatus` (which is an enum) to update the status of an MLflow run

Comment (smurching) [reply mlflow/store/artifact/unity_catalog_models_artifact_repo.py:29]: Here's [the client logic](https://github.com/mlflow/mlflow/blob/2e9c982b6f9a010c744b2e3d1a3a1c1203538cb3/mlflow/tracking/_tracking_service/client.py#L357) that calls the store `update_run` method, I guess it just converts the value to an int explicitly?

Comment (jerrylian-db) [reply mlflow/store/artifact/unity_catalog_models_artifact_repo.py:29]: The output of this command
```
message_to_json(
  UpdateRun(
      run_uuid="1234",
      run_id="1234",
      status=RunStatus.from_string("RUNNING"),
      end_time=1,
      run_name="hello world",
  )
)
```
is this `'{\n  "run_uuid": "1234",\n  "status": "RUNNING",\n  "end_time": 1,\n  "run_id": "1234",\n  "run_name": "hello world"\n}'`

My guess is that the MLflow backend handles the enums differently. Here's how the `UpdateRun` handler in MLflow uses the status
```
trackingDatabase.updateRun(
  orgId = orgId,
  experimentId,
  uuid = uuid,
  status = rpc.getStatus,
  endTime = rpc.getEndTime
)
```
Where `rpc.getStatus` simple returns the `RunStatus` enum. In the uc-model-registry, we do this:
```
rpc.operation.map(x => ModelVersionOperation.fromValue(x.id))
```
Yeah, not sure what the difference is here. When I tested sending the `operation` field as a string with CURL, it didn't work.

Comment (smurching) [reply mlflow/store/artifact/unity_catalog_models_artifact_repo.py:29]: > Yeah, not sure what the difference is here. When I tested sending the operation field as a string with CURL, it didn't work.

What was the error out of curiosity (let's discuss that outside of the OSS github thread)? In MlflowServiceSuite, we have some integration tests that verify we can pass a `status` to UpdateRun as a string - I wonder if we could write a similar test for the UC backend to quickly debug & see what's going on

Comment (jerrylian-db) [reply mlflow/store/artifact/unity_catalog_models_artifact_repo.py:29]: Result of discussion: we will rename the OSS enum names.

**Comment by jerrylian-db** [comment]
@mlflow-automation autoformat

**Comment by jerrylian-db** [comment]
@mlflow-automation autoformat

**Review by smurching — APPROVED** [review]
LGTM!

**Pull request merged**